calculator.js

```diff
@@ -1,27 +1,124 @@
+  num1: 3,
+  num2: 5,
+  operation: '+',
+}
+const exampleSubtractionInput1 = {
+  num1: 3,
+  num2: 5,
+  operation: 'subtract',
+}
+const exampleSubtractionInput2 = {
+  num1: 3,
+  num2: 5,
+  operation: '-',
+}
+const exampleMultiplicationInput1 = {
+  num1: 3,
+  num2: 5,
+  operation: 'multiply',
+}
+const exampleMultiplicationInput2 = {
+  num1: 3,
+  num2: 5,
+  operation: '*',
+}
+const exampleDivisionInput1 = {
+  num1: 3,
+  num2: 5,
+  operation: 'divide',
+}
+const exampleDivisionInput2 = {
+  num1: 3,
+  num2: 5,
+  operation: '/',
+}
+
+const translateOperator = function(promptInput) {
+  const validOperators = ['+', '-', '*', '/'];
+  promptInput.operation = promptInput.operation.toLowerCase();
+  switch(promptInput.operation) {
+    case 'add':
+      promptInput.operation = '+';
+      break;
+    case 'subtract':
+      promptInput.operation = '-';
+      break;
+    case 'multiply':
+      promptInput.operation = '*';
+      break;
+    case 'divide':
+      promptInput.operation = '/';
+      break;
+  }
+
+  if (validOperators.includes(promptInput.operation)) {
+    return promptInput;
+  } else {
+   console.log('unsupported operator');
+   return null;
+  }
+}
 
 const calculateUserInput = function (error, promptInput) {
-  console.log('This is the value of the promptInput variable that got passed in by prompt, after our prompt package collect user input', promptInput);
-
-  // Questions to ask and answer:
-  // What is promptInput?
-  // What data type? What does it hold? What does it represent?
-  // How do we read values from it? What syntax?
-  // How can we use it?
-  // Can we call our existing functions now, inside of this function?
-}
-
-// Example manual testing of calculator.  
-// calculateUserInput({}, {
-//   num1: 3,
-//   num2: 4,
-//   operation: 'add',
-// });
+  // console.log('This is the value of the promptInput variable that got passed in by prompt, after our prompt package collect user input', promptInput);
+
+  if (promptInput.num1 != Number(promptInput.num1)) {
+    console.log(`${promptInput.num1} is not a number.`)
+    return null;
+  } else if (promptInput.num2 != Number(promptInput.num2)) {
+    console.log(`${promptInput.num2} is not a number.`)
+    return null;
+  }
+
+  promptInput.num1 = Number(promptInput.num1);
+  promptInput.num2 = Number(promptInput.num2);
+  if (translateOperator(promptInput) == null) {
+    return null;
+  }
+
+  let result;
+  switch(promptInput.operation) {
+    case '+':
+      result = promptInput.num1 + promptInput.num2;
+      break;
+    case '-':
+      result =  promptInput.num1 - promptInput.num2;
+      break;
+    case '*':
+      result = promptInput.num1 * promptInput.num2;
+      break;
+    case '/':
+      if (promptInput.num2 == 0) {
+        console.log('You cannot divide by zero');
+      } else {
+        result = promptInput.num1 / promptInput.num2;
+      }
+      break;
+    default:
+      throw 'Error occurred. Program exited.';
+  }
```

Comment:
If you put this `switch` into its own helper function you can just use `return` and then avoid using any `let`s.